File: src/main/java/net/database/TownsAPI.java
package net.database;

import client.Character;

import static database.jooq.Tables.UNLOCKEDTOWNS;

public class TownsAPI {

    public static void load(Character chr) {
        DatabaseCore.getConnection()
                .select(UNLOCKEDTOWNS.TOWN).from(UNLOCKEDTOWNS)
                .where(UNLOCKEDTOWNS.CID.eq(chr.getId()))
                .fetch()
                .forEach(rec -> chr.getTowns().add(rec.getValue(UNLOCKEDTOWNS.TOWN)));
    }

    public static void add(Character chr, int town) {
        DatabaseCore.getConnection()
                .insertInto(UNLOCKEDTOWNS, UNLOCKEDTOWNS.CID, UNLOCKEDTOWNS.TOWN)
                .values(chr.getId(), town)
                .execute();
    }
}

File: src/main/java/net/database/TownsAPI.kt
package net.database

import client.Character
import database.jooq.Tables.UNLOCKEDTOWNS
import net.database.DatabaseCore.connection

object TownsAPI {

    fun load(chr: Character) {
        with(UNLOCKEDTOWNS) {
            connection.select(TOWN).from(this)
                .where(CID.eq(chr.id))
                .fetch()
                .forEach { chr.towns.add(it.getValue(TOWN)) }
        }
    }

    fun add(chr: Character, town: Int) {
        with(UNLOCKEDTOWNS) {
            connection.insertInto(this, CID, TOWN)
                .values(chr.id, town)
                .execute()
        }
    }
}